src/mod.ts

```diff
@@ -0,0 +1,128 @@
+export type {
+  Manifest,
+  ManifestWasm,
+  ManifestWasmUrl,
+  ManifestWasmData,
+  ManifestLike as IntoManifest
+} from './manifest.ts';
+import { FEATURES } from 'js-sdk:features';
+export { FEATURES } from 'js-sdk:features';
+
+import {
+  ManifestLike,
+  intoManifest as _intoManifest,
+  toWasmModuleData as _toWasmModuleData
+} from './manifest.ts';
+
+import { createForegroundPlugin as _createForegroundPlugin } from './foreground-plugin.ts';
+import { createBackgroundPlugin as _createBackgroundPlugin } from './background-plugin.ts';
+export { type CallContext } from './call-context.ts';
+
+/**
+ * {@link ExtismPlugin} Config
+ */
+export interface PluginConfigLike {
+  [key: string]: string
+}
+export type PluginConfig = { [key: string]: string };
+
+/**
+ * Options for initializing an Extism plugin.
+ */
+export interface ExtismPluginOptions {
+  useWasi?: boolean | undefined;
+  offMainThread?: boolean | undefined;
+  logger?: Console;
+  functions?: { [key: string]: { [key: string]: any } } | undefined;
+  allowedPaths?: { [key: string]: string } | undefined;
+  allowedHosts?: string[] | undefined;
+  config?: PluginConfigLike | undefined;
+  fetch?: typeof fetch;
+}
+
+export interface Plugin {
+  /**
+   * Check if a function exists in the WebAssembly module.
+   *
+   * @param {string | [string, string]} funcName The function's name, or a tuple of target module name and function name.
+   * @returns {Promise<boolean>} true if the function exists, otherwise false
+   */
+  functionExists(funcName: string | [string, string]): Promise<boolean>;
+  close(): Promise<void>;
+
+  /**
+   * Call a specific function from the WebAssembly module with provided input.
+   *
+   * @param {string | [string, string]} funcName The name of the function to call
+   * @param {Uint8Array | string} input The input to pass to the function
+   * @returns {Promise<Uint8Array>} The result from the function call
+   */
+  call(funcName: string | [string, string], input?: string | number | Uint8Array): Promise<Uint8Array | null>;
+  callBlock(funcName: string | [string, string], input?: number | null): Promise<[number | null, number | null]>;
+  getExports(name?: string): Promise<WebAssembly.ModuleExportDescriptor[]>;
+  getImports(name?: string): Promise<WebAssembly.ModuleImportDescriptor[]>;
+  getInstance(name?: string): Promise<WebAssembly.Instance>;
+}
+
+// TODO: move these into another module...
+export interface InternalConfig {
+  logger: Console,
+  allowedHosts: string[],
+  allowedPaths: {[key: string]: string},
+  functions: {[namespace: string]: {[func: string]: any}},
+  fetch: typeof fetch,
+  wasiEnabled: boolean,
+  config: PluginConfig,
+}
+
+export interface InternalWasi {
+  importObject(): Promise<Record<string, WebAssembly.ImportValue>>;
+  initialize(instance: WebAssembly.Instance): Promise<void>
+}
+
+export default async function createPlugin(
```

Comment:
Nit: we tend to lean towards `newPlugin` in the other SDKs

Reply:
I'll export `newPlugin` as an alias of `createPlugin` if that's okay! (I went with `createPlugin` to follow the `create*` pattern from the web platform/node; like `URL.createObjectURL`, `fs.createReadStream`, and `http.createServer`.)

Reply:
I'm good with just leaving create then. Less to document!